packages/contracts/src/contracts/current/protocol/AssetProxyDispatcher/AssetProxyDispatcher.sol

```diff
@@ -0,0 +1,85 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.21;
+
+import "./IAssetProxyDispatcher.sol";
+import "./IAssetProxy.sol";
+import "../../utils/Ownable/Ownable.sol";
+import "../../utils/Authorizable/Authorizable.sol";
+
+contract AssetProxyDispatcher is
+    IAssetProxyDispatcher,
+    Ownable,
+    Authorizable
+{
+    // Mapping from Asset Proxy Id's to their respective Asset Proxy
+    mapping (uint8 => IAssetProxy) public assetProxies;
+
+    /// @dev Transfers assets. Either succeeds or throws.
+    /// @param assetMetadata Byte array encoded for the respective asset proxy.
+    /// @param from Address to transfer token from.
+    /// @param to Address to transfer token to.
+    /// @param amount Amount of token to transfer.
+    function transferFrom(
+        bytes assetMetadata,
+        address from,
+        address to,
+        uint256 amount)
+        external
+        onlyAuthorized
+    {
+        // Lookup asset proxy
+        require(assetMetadata.length >= 1);
+        uint8 assetProxyId = uint8(assetMetadata[0]);
+        IAssetProxy assetProxy = assetProxies[assetProxyId];
+
+        // transferFrom will either succeed or throw.
+        assetProxy.transferFrom(assetMetadata, from, to, amount);
+    }
+
+    /// @dev Adds a new asset proxy.
+    /// @param assetProxyId Id of the asset proxy.
+    /// @param newAssetProxy Asset proxy contract to add.
+    /// @param currentAssetProxy Existing asset proxy to overwrite.
+    function addAssetProxy(
```

Comment:
It might be YAGNI, but how do we feel about an human+contract readable string associated with what the Proxy is? Similar to the Token Registry + Symbol? I.e the ERC20 Token Proxy would have `id: 1, name: 'ERC20'`. 

I haven't fully formed this thought yet, but a forwarding contract might want overloaded functions (i.e buy anything), but needs to perform transfer logic. So it might need a way to look up, `if (lookup(firstByte(makerMetadata)) == 'ERC721')`, therefor call `safeTransfer`.

OTOH the forwarding contract could always set allowances before transfer (erc721) and use the DelegateProxy. 

Maybe it's YAGNI...